Do not make symbols hidden on Blackfin

The libselinux logic to hide internal symbols from the DSO doesn't
work properly on Blackfin due to the USER_LABEL_PREFIX not being
handled properly. A real fix is not that simple, so this patch simply
disables the internal symbol hiding mechanism. This means that those
symbols are visible in the final DSO, which is not a problem for
proper execution, it just isn't as clean.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Signed-off-by: Adam Duskett <Aduskett@gmail.com>

Index: b/src/dso.h
===================================================================
--- a/src/dso.h
+++ b/src/dso.h
@@ -1,7 +1,7 @@
 #ifndef _SEPOL_DSO_H
 #define _SEPOL_DSO_H	1
 
-#ifdef SHARED
+#if defined(SHARED) && !defined(__bfin__)
 # define hidden __attribute__ ((visibility ("hidden")))
 # define hidden_proto(fct) __hidden_proto (fct, fct##_internal)
 # define __hidden_proto(fct, internal)	\
